packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider.py

```
@@ -0,0 +1,147 @@
+                    config.token_url,
+                    config.scopes,
+                    config.service_token is not None)
+
+    def _load_function(self, function_path: str):
+        """Load a Python function from a module path string (e.g., 'my_module.get_token')."""
+        try:
+            module_name, func_name = function_path.rsplit(".", 1)
+            module = importlib.import_module(module_name)
+            func = getattr(module, func_name)
+            logger.info("Loaded service token function: %s", function_path)
+            return func
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)